mirage-channel.opam

```diff
@@ -7,7 +7,7 @@ homepage: "https://github.com/mirage/mirage-channel"
 doc: "http://mirage.github.io/mirage-channel/"
 bug-reports: "https://github.com/mirage/mirage-channel/issues"
 depends: [
-  "ocaml" {>= "4.06.0"}
+  "ocaml" {>= "4.07.0"}
```

Comment:
What syntax error do you have (just to keep a record if we see the same message in another context)

Reply:
```
/src: (run (shell "opam exec -- dune build @install @check @runtest && rm -rf _build"))
File "test/test_channel.ml", line 73, characters 4-8:
Error: Syntax error
"/bin/bash" "-c"
"opam exec -- dune build @install @check @runtest && rm -rf _build" failed with exit status 1
```

The contents of `test_channel.ml`:
```OCaml
    type flow = unit
    type error = |
    type write_error = Mirage_flow.write_error
```

I suspect it is the `type error = |`.